Reminder-App/Reminder-App/ReminderListViewController.swift

```diff
@@ -9,32 +9,217 @@ import UIKit
+                            
+                            alert.addAction(UIAlertAction(title: actionTitle,
+                                                          style: .default,
+                                                          handler: {
+                                                            _ in
+                                                            self.dismiss(animated: true, completion: nil)
+                                                          }))
+                            
+                            self.present(alert, animated: true, completion: nil)
+                        }
+                    }
+                }
+            })
         }
     }
     
     // MARK: Lifecycle Methods
     override func viewDidLoad() {
-        
         super.viewDidLoad()
         
-        reminderListDataSource = ReminderListDataSource()
+        reminderListDataSource = ReminderListDataSource(
+            reminderCompletedAction: {
+                reminderIndex in
+                DispatchQueue.main.async {
+                    self.tableView.reloadRows(at: [IndexPath(row: reminderIndex, section: 0)], with: .automatic)
+                    self.refreshProgressView()
+                }
+            }, reminderDeletedAction: {
+                DispatchQueue.main.async {
+                    self.refreshProgressView()
+                }
+            }, remindersChangedAction: {
+                DispatchQueue.main.async {
+                    self.tableView.reloadData()
+                    self.refreshProgressView()
+                }
+            })
+        
         tableView.dataSource = reminderListDataSource
     }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        let radius = view.bounds.size.width * 0.5 * 0.7
+        progressContainerView.layer.cornerRadius = radius
+        progressContainerView.layer.masksToBounds = true
+        self.refreshProgressView()
+        refreshBackground()
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        
+        if let navigationController = navigationController,
+           navigationController.isToolbarHidden {
+            navigationController.setToolbarHidden(false, animated: animated)
+        }
+    }
+    
+    // MARK: Private Methods
+    private func addReminder() {
+        let storyboard = UIStoryboard(name: Self.mainStoryboardName, bundle: nil)
+        let detailViewController: ReminderDetailViewController = storyboard.instantiateViewController(identifier: Self.detailViewControllerIdentifier)
+        let reminder = Reminder(id: UUID().uuidString, title: "New Reminder", dueDate: Date())
+        
+        detailViewController.configure(with: reminder,
+                                       isNew: true,
+                                       addAction: {
+                                        reminder in
```

Comment:
if addAction is last closure / Trailing Closure then you can omit its name like 
 detailViewController.configure(with: reminder,
                                       isNew: true) { // Closure of addAction}

Reply:
Whenever i do that the compiler gives me this warning:
"Backward matching of the unlabeled trailing closure is deprecated; label the argument with 'editAction' to suppress this warning"
So thats why i didn't omit the label.